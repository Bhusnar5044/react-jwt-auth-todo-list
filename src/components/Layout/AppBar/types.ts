export type Props = {
    // Header prop type
    pages: { label: string; path: string }[];
};
