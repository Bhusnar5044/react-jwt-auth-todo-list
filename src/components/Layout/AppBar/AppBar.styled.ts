import { theme } from '@theme';
import { NavLink } from 'react-router-dom';
import { styled } from 'styled-components';

export const HeaderContainer = styled('header')`
    display: flex;
    align-items: center;
    width: 100%;
    margin: auto;
    background-color: ${theme.colors.success.main};
    padding: 1rem 0;
`;

export const HeaderContent = styled.div`
    display: flex;
    align-items: center;
    justify-content: space-between;
    width: 90%;
    margin: auto;
    color: #fff;
    a:hover {
        color: ${theme.colors.info.main};
    }
`;

export const Logo = styled.img`
    width: 2.5rem;
    height: auto;
`;

export const StyledLink = styled(NavLink)<{ activeClassName?: string }>`
    color: #fff;
    :hover {
        color: ${theme.colors.info.main};
    }
    &.active {
        color: ${theme.colors.info.main};
    }
`;
