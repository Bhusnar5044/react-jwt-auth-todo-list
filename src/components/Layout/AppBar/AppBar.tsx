import { Box, Text } from '@components/Lib';
import { useAuth } from '@context/AuthProvider';
import { FC, memo } from 'react';
import { HeaderContainer, HeaderContent, Logo, StyledLink } from './AppBar.styled';

export const AppBar: FC = memo(() => {
    const { tokens, logout } = useAuth();
    const handleLogout = () => {
        logout?.();
    };
    return (
        <HeaderContainer>
            <HeaderContent>
                <Box gap="2rem" alignItems="center">
                    <Logo src="/vite.svg" />
                    <Text>React JWT Demo</Text>
                </Box>
                {tokens?.access_token && (
                    <Box gap="2rem" alignItems="center">
                        <StyledLink to={'/'} activeClassName={location.pathname === `/` ? 'active' : ''}>
                            <Text>Dashboard</Text>
                        </StyledLink>
                        <StyledLink to={'/profile'} activeClassName={location.pathname === `/profile` ? 'active' : ''}>
                            <Text>Profile</Text>
                        </StyledLink>

                        <Text as={'a'} onClick={handleLogout}>
                            Logout
                        </Text>
                    </Box>
                )}
            </HeaderContent>
        </HeaderContainer>
    );
});

AppBar.displayName = 'AppBar';
