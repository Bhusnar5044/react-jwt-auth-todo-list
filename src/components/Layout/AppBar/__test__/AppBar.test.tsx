import { renderWithRouter, screen } from '@test-utils';
import { AppBar } from '../AppBar';

describe('AppBar', () => {
    it('should render properly', () => {
        const { container } = renderWithRouter(<AppBar />);
        expect(container).toMatchSnapshot();
        expect(screen.getByRole('img')).toBeInTheDocument();
    });
});
