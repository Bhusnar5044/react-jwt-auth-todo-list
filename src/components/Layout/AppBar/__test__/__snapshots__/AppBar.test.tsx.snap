// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar should render properly 1`] = `
.c2 {
  display: flex;
  align-items: center;
  gap: 2rem;
}

.c4 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 400;
  letter-spacing: 0rem;
  line-height: 2.2rem;
  text-align: initial;
  display: inline-block;
}

.c0 {
  display: flex;
  align-items: center;
  width: 100%;
  margin: auto;
  background-color: #008085;
  padding: 1rem 0;
}

.c1 {
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 90%;
  margin: auto;
  color: #fff;
}

.c1 a:hover {
  color: #495057;
}

.c3 {
  width: 2.5rem;
  height: auto;
}

<div>
  <header
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        alignitems="center"
        class="c2"
        gap="2rem"
      >
        <img
          class="c3"
          src="/vite.svg"
        />
        <span
          class="c4"
          textalign="initial"
          textcolor="inherit"
        >
          React JWT Demo
        </span>
      </div>
    </div>
  </header>
</div>
`;
